README.md

```diff
@@ -19,10 +19,81 @@ Create a `BanyanDBClient` with host, port and a user-specified group and then es
 ```java
 // use `default` group
 client = new BanyanDBClient("127.0.0.1", 17912, "default");
-// establish a connection
+// to send any request, a connection to the server must be estabilished
 client.connect(channel);
 ```
 
+## Schema Management
+
+### Stream
+
+Then we may define a stream with customized configurations,
+
+```java
+// build a stream "sw" with 2 shards and ttl equals to 30 days
+Stream s = new Stream("sw", 2, Duration.ofDays(30));
+s.addTagNameAsEntity("service_id").addTagNameAsEntity("service_instance_id").addTagNameAsEntity("state");
+// TagFamily - data
+TagFamilySpec dataFamily = new TagFamilySpec("data");
+dataFamily.addTagSpec(TagFamilySpec.TagSpec.newBinaryTag("data_binary"));
+s.addTagFamilySpec(dataFamily);
+// TagFamily - searchable
+TagFamilySpec searchableFamily = new TagFamilySpec("searchable");
+searchableFamily.addTagSpec(TagFamilySpec.TagSpec.newStringTag("trace_id"))
+        .addTagSpec(TagFamilySpec.TagSpec.newIntTag("state"))
+        .addTagSpec(TagFamilySpec.TagSpec.newStringTag("service_id"));
+s.addTagFamilySpec(searchableFamily);
+// create with the stream schema, client is the BanyanDBClient created above
+s = client.define(s);
+```
+
+where simple APIs (i.e. `BanyanDBClient.define`) are provided to define the schema of `Measure` or `Stream`.
+
+### IndexRules
+
+For better search performance, index rules are necessary for `Stream` and `Measure`. You have to
+specify a full list of index rules that would be bounded to the target `Stream` and `Measure`.
+
+```java
+// create IndexRule with inverted index type and save it to series store
+IndexRule indexRule = new IndexRule("db.instance", IndexRule.IndexType.INVERTED, IndexRule.IndexLocation.SERIES);
+// tag name specifies the indexed tag
+indexRule.addTag("db.instance");
+// create the index rule "db.instance"
+client.defineIndexRules(stream, beginAt, expireAt, indexRule);
+```
+
+where multiple index rules are supported. Internally, an `IndexRuleBinding` is created automatically for users,
+which will be active between `beginAt` and `expireAt`.
+
+### Measure
+
+`Measure` can also be defined directly with `BanyanDBClient`,
+
+```java
+// create a measure registry
+// create a new measure schema with 2 shards and ttl 30 days.
+Measure m = new Measure("measure-example", 2, Duration.ofDays(30));
+// set entity
+m.addTagNameAsEntity("service_id").addTagNameAsEntity("service_instance_id").addTagNameAsEntity("state");
+// add tag family: "searchable"
+TagFamilySpec searchableFamily = new TagFamilySpec("searchable");
+searchableFamily.addTagSpec(TagFamilySpec.TagSpec.newStringTag("trace_id"))
+    .addTagSpec(TagFamilySpec.TagSpec.newIntTag("state"))
+    .addTagSpec(TagFamilySpec.TagSpec.newStringTag("service_id"));
+m.addTagFamilySpec(searchableFamily);
+// set interval rules
+m.addIntervalRule(Measure.IntervalRule.matchStringLabel("interval", "day", "1d"));
+m.addIntervalRule(Measure.IntervalRule.matchNumericLabel("interval", 3600L, "1h"));
```

Comment:
@hanahmily How does this rule work with multiple-dimensionality metrics in one measure?

Reply:
Supporting a label `scope` which indicates the dimension, its candidates are `day`, `hour`, and `minute`. The rule should be:

```java 
m.addIntervalRule(Measure.IntervalRule.matchStringLabel("scope", "day", "1d"));
m.addIntervalRule(Measure.IntervalRule.matchStringLabel("scope", "hour", "1h"));
m.addIntervalRule(Measure.IntervalRule.matchStringLabel("scope", "minute", "1m"));
```

The value expression supports a decimal number with a unit suffix.

Reply:
Got it. @lujiajing1126 Please use @hanahmily 's case to replace yours. It is much better.

Reply:
Suggestion applied